Adding first draft of family history table

rosibaj opened the pull request:
- draft of family history table from extended excel
- Edited some descriptions from google doc for consistent verbiage with dictionary

Questions after making table: 

1. should `relative_death_status` mimic `donor.cause_of_death` more to be consistent with the donor table? 
From 
```
    {
      "description": "Indicate if the patient's relative died from the cancer they were diagnosed with.",
      "name": "relative_death_status",
      "restrictions": {
        "codeList": "#/list/yes_no"
      },
      "valueType": "string",
      "displayName": "Death Status Of Relative"
    },
 ```
to 
```
   {
      "description": "Indicate the cause of the death of the relative.",
      "name": "cause_of_death_of_relative",
      "restrictions": {
        "codeList": ["Died of cancer", "Died of other reasons", "Unknown"],
      },
      "valueType": "string",
      "meta": {
        "core": true,
        "displayName": "Cause of Death of Relative "
      }
    }
 ```

2: Re: References.  @hknahal would you like to add an additional column to the dictionary UI at https://docs.icgc-argo.org/ ?  We could add a whole `meta` field for `source data references` or something like that to support this?

hknahal commented:
> 1. should `relative_death_status` mimic `donor.cause_of_death` more to be consistent with the donor table?

@rosibaj Yes good point, I agree. We should change it to "cause_of_death_of_relative" instead. 

> 2: Re: References. @hknahal would you like to add an additional column to the dictionary UI at https://docs.icgc-argo.org/ ? We could add a whole `meta` field for `source data references` or something like that to support this?

For now, we've been adding references to standards in the "Field & Description" column, but let me think about this suggestion and get back to you.

hknahal commented on schemas/family_history.json:
```diff
@@ -0,0 +1,144 @@
+      "displayName": "Age Of Relative At Diagnosis"
+    },
+    {
+      "name": "disease_icd10_of_relative",
```
Should we re-name this field to `cancer_type_code_of_relative` so it's consistent with the field name we use in the Primary Diagnosis table (`cancer_type_code`)?

rosibaj replied:
good catch and updated!

hknahal commented on schemas/family_history.json:
```diff
@@ -0,0 +1,144 @@
+    {
+      "description": "The age (in years) when the patient's relative was first diagnosed. (Reference: caDSR CDE ID 5300571)",
+      "name": "age_of_relative_at_diagnosis",
+      "restrictions": {
```
I think this codelist got mistakenly inserted here.

hknahal commented on schemas/family_history.json:
```diff
@@ -0,0 +1,140 @@
+  "name": "family_history",
+  "description": "The collection of data elements related to a specific donor in an ARGO program.",
+  "meta": {
+    "parent": "specimen"
```
Is this supposed to be `donor`?

rosibaj commented:
@hknahal Asked fore re-review: 
- fixed review comments 
- Added relative id as discussed to keep records unique 
- updated description 
- fixed json formatting

hknahal commented on schemas/family_history.json:
```diff
@@ -0,0 +1,154 @@
+        ]
+      },
+      "valueType": "string",
+      "displayName": "Gender of Relatives"
```
Make "Relatives" singular since we're asking the gender of just one relative (ie. not an array field).

hknahal commented on schemas/family_history.json:
```diff
@@ -0,0 +1,154 @@
+      "valueType": "string",
+      "meta": {
+        "core": true,
+        "displayName": "Cause of Death of Relative "
```
Extra whitespace at end of "Cause of Death of Relative "

hknahal reviewed (approved):
@rosibaj I noticed just a few really small things (mentioned in comments above) but otherwise it looks fine.
